src/offline/cable_parameters.F90

```diff
@@ -2740,11 +2756,660 @@ SUBROUTINE init_veg_from_vegin(ifmp,fmp, veg, soil_zse )
+             insucs(landpt(e)%ilon,landpt(e)%ilat)
+          soil%bch_vec(landpt(e)%cstart:landpt(e)%cend,klev) =& 
+             inbch(landpt(e)%ilon,landpt(e)%ilat)
+          soil%rhosoil_vec(landpt(e)%cstart:landpt(e)%cend,klev) =& 
+             inrhosoil(landpt(e)%ilon,landpt(e)%ilat)
+          soil%cnsd_vec(landpt(e)%cstart:landpt(e)%cend,klev) =& 
+             incnsd(landpt(e)%ilon,landpt(e)%ilat)
+          soil%css_vec(landpt(e)%cstart:landpt(e)%cend,klev) =& 
+             incss(landpt(e)%ilon,landpt(e)%ilat)
+          end do
+       end do
+   end if
+    ! _______ MMY @Oct2022 taken from write_default_params _______
+    soil%elev       = max(0.0,min(1.0e10,soil%elev))
+    soil%slope      = max(0.000001,min(0.9,soil%slope))
+    soil%slope_std  = max(0.0005,min(0.01,soil%slope_std))
+    soil%GWdz       = max(7.0,min(150.0,soil%GWdz))
+    soil%drain_dens = max(1.0e-8,min(0.01,soil%drain_dens))
+    ! ____________________________________________________________
+    
+    ! ????????? MMY move from derived_parameters, but should it be here ?????????
+!    print *,'MMY-phase 2 check, soil_depth(:,klev)', soil_depth(:,klev)
+!    print *,'MMY-phase 2 check, soil%zdepth(:)', soil%zdepth(:)
+!    print *,'MMY-phase 2 check, soil%hkrz(:)', soil%hkrz(:)
+!    print *,'MMY-phase 2 check, soil%hyds_vec(:,klev)', soil%hyds_vec(:,klev)
+
+! 2 lines below inserted to fix bug of uninitialized soil%hkrz(:) and soil%zdepth(:) here in MMY code - rk4417 - phase2
+! 2 lines below copied from SUBROUTINE derived_parameters which is called after SUBROUTINE GWspatialParameters - rk4417 - phase2
+    
+    soil%hkrz(:)    = real(gw_params%hkrz,r_2)
+    soil%zdepth(:)  = real(gw_params%zdepth,r_2)
+    
+      DO klev=1,ms
+         soil%hyds_vec(:,klev) = soil%hyds_vec(:,klev)*exp(-soil%hkrz(:)*(soil_depth(:,klev)-soil%zdepth(:)))
+      END DO
+    ! ???????????????????????????????????????????????????????????????????????????
+    !removed gw_soils for now
+  
+  END SUBROUTINE GWspatialParameters
+
+! below to the end of the module added by rk4417 - phase2
+  
+   function get_gw_2d_var_constdef(ncfile_id,try_it,varname,default_const,nlon,nlat) result(data_vec)
+         use netcdf
+         use cable_def_types_mod, only : r_2,mp
+         real, intent(in) :: default_const
+         integer, intent(in) :: try_it,nlon,nlat
+         character(len=*), intent(in) :: varname
+         integer, intent(in)               :: ncfile_id
+         real(r_2), dimension(mp) :: data_vec
+
+         real, dimension(nlon,nlat) :: GW2d_data
+         real, dimension(nlon,nlat) :: default_data
+         integer :: i,j,k,n
+         integer :: varid
+         integer :: varinq_status,varget_status
+
+         print *, "MMY in get_gw_2d_var_constdef" ! MMY add to test whether get_gw_2d_var_constdef is necessary
+         default_data(:,:) = default_const
+
+         if (.not.cable_user%gw_model) then
+
+            GW2d_data(:,:) = default_data(:,:)
+
+         else
```

Comment:
I would like to remove this case. This should be done outside these routines. Keep these generic.